solr/core/src/java/org/apache/solr/search/facet/FacetMerger.java

```diff
@@ -44,13 +44,14 @@ public Map<String, Object> getRefinement(Context mcontext) {
   public static class Context {
     // FacetComponentState state;  // todo: is this needed?
     final int numShards;
-    // [bucket0_shard0, bucket0_shard1, bucket0_shard2,  bucket1_shard0,
-    private final BitSet sawShard = new BitSet();
+    // BitSet per shard, each bit set holds bucket nums for whether or not a bucket has been seen for a given shard.
+    private final BitSet[] sawShard;
```

Comment:
Thought about using Solr's `LongBitSet` but you have to preallocate / grow manually.
Honestly, feels like this approach will keep each bit set much smaller because we are no longer multiplying by numShards for each bucket.